src/lib/jsonSource/jsonSourceClient.js

```diff
@@ -0,0 +1,60 @@
+import promises from "../../../public/promises.json";
+
+import { slugify } from "@/promisetracker/utils";
+
+const JsonSourceClient = () => {
+  // Sulugify all promises
+  let allPromises = promises.promises;
+
+  allPromises = allPromises.map((promise) => {
+    const slug = slugify(promise.title);
+    return {
+      ...promise,
+      href: `/promises/${promise.id}/${slug}`,
+      slug,
+    };
+  });
+
+  return {
+    query({ query, limit, category, slug }) {
+      switch (query) {
+        case "GET_PROMISES":
+          return allPromises.slice(0, limit);
+        case "GET_PROMISE":
+          try {
+            const promise = allPromises.filter((p) => `${p.slug}` === slug)[0];
+            return promise;
+          } catch (e) {
+            return [];
+          }
+        case "GET_KEY_PROMISES": {
+          const keyPromises = allPromises.filter((p) => p.keyPromise) ?? [];
+          return keyPromises.slice(0, limit);
```

Comment:
Can chain these...
```suggestion
          const keyPromises = allPromises.filter((p) => p.keyPromise).slice(0, limit);
          return keyPromises;
```

filter will return empty array if no match was found I believe.